Add DockerMock class

nre-ableton opened the pull request:
This class makes it possible to mock the `docker` singleton, which will make it possible for us to test much more of our pipeline code. Also this PR includes some groovylint cleanups, since it turns out that RTB was not configured to automatically build this repo on PR/push events.

ptal @AbletonDevTools/gotham-city, thanks!

ala-ableton commented on src/main/groovy/com/ableton/JenkinsMocks.groovy:
```diff
@@ -80,8 +80,8 @@ class JenkinsMocks {
     } else if (returnStatus) {
       return output.exitValue
-    } else {
-      return output.exitValue == 0
```
Please change the above `else if` into a simple `if` as well.

nre-ableton replied:
@ala-ableton Done!

ala-ableton replied:
I was expecting a single commit with both changes (with a `Do not use else after return` kind of message), but the current commits are fine as well.

nre-ableton committed:
Cleanup: place static imports first

nre-ableton committed:
Cleanup: remove unnecessary else statement

nre-ableton committed:
Introduce class: DockerMock

This class of mostly noop functions can be used to mock the docker
singleton from the Jenkins Docker Workflow plugin.

nre-ableton committed:
Cleanup: refactor else if statement to just if

nre-ableton committed:
Cleanup: remove unnecessary setUp() override

nre-ableton committed:
Cleanup: prefer single-quoted strings

ala-ableton commented:
LGTM :+1:
